app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,60 @@
+class PostsController < ApplicationController
+  before_action :get_post, only: [:show, :destroy, :edit, :update]
+  before_action :get_comments, only: [:show]
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def show
+  end
+
+  def create
+    @post = Post.new(post_params)
+    if @post.save
+      flash[:notice] = 'Post succesfully saved'
+      redirect_to post_path(@post)
+    else
+      flash[:warn] = 'Your post sucks. Please try again'
+      redirect_to new_post_path
```

Comment:
I would render :new rather than redirect here. Then you can have the user's post content still in the form and perhaps show the errors from the @post object that prevented it from saving. If you redirect the browser requests a new page and the user's input is lost. And then they hate you. Especially if you just told them their post sucks.